src/github.com/getlantern/flashlight/ui/ui.go

```diff
@@ -112,13 +112,31 @@ func Show() {
 		if err != nil {
 			log.Errorf("Error opening page to `%v`: %v", uiaddr, err)
 		}
-		if externalUrl != "NO"+"_URL" && !openedExternal {
-			time.Sleep(4 * time.Second)
-			err = open.Run(externalUrl)
-			if err != nil {
-				log.Errorf("Error opening external page to `%v`: %v", uiaddr, err)
-			}
-			openedExternal = true
-		}
+		openExternalUrl()
+	}()
+}
+
+// openExternalUrl opens an external URL of one of our partners automatically
+// at startup if configured to do so. It should only open the first time in
+// a given session that Lantern is opened.
+func openExternalUrl() {
+	if openedExternal {
+		log.Debugf("Not opening external URL again")
+		return
+	}
+	defer func() {
```

Comment:
Strictly speaking this `openedExternal` variable will be set to `true` even if the page could not be opened and we'll also have this 4s period on which `Show()` will continue opening browser windows by calling `openExternalUrl()`. That's not a big deal but maybe we could try setting `openedExternal` to `true` immediately after calling `openExternalUrl()` and only set if to `false` if something failed.

```
func openExternalUrl() {
  if openedExternal {
    log.Debugf("Not opening external URL again")
    return
  }

  openedExternal = true

  path, s, err := packaged.ReadSettings()
  if err != nil {
    openedExternal = false
    log.Errorf("Could not read yaml from %v: %v", path, err)
    return
  }

  time.Sleep(4 * time.Second)
  err = open.Run(s.StartupUrl)
  if err != nil {
    openedExternal = false
    log.Errorf("Error opening external page to `%v`: %v", uiaddr, err)
  }
}
```